src/app/shared/services/authentication.service.ts

```diff
@@ -1,34 +1,54 @@
 import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
 
-import { User } from '../interfaces/user.interface.';
+import { BehaviorSubject, Observable, Subject, last } from 'rxjs';
+
+import { HttpUser } from '../interfaces/http-user.interface';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthenticationService {
 
-  loggedUser!: User | null;
-
-  login(user: User): void {
-    this.loggedUser = {...user};
-    localStorage.setItem("token", JSON.stringify(this.loggedUser));
-    window.location.reload();
+  loggedUser!: HttpUser | null;
+  private apiUrl = 'http://localhost:3004';
+  userSubject = new Subject<HttpUser>();
+  isAuthSubject = new BehaviorSubject<boolean>(false);
```

Comment:
I suppose it should be private
Please fix the order
Public-protected-private properties

Reply:
the order was fixed